spirit/user/perm/decorators.py

```diff
@@ -0,0 +1,160 @@
+                path = request.get_full_path()
+            from django.contrib.auth.views import redirect_to_login
+            return redirect_to_login(
+                path, resolved_login_url, redirect_field_name)
+        return _wrapped_view
+    return decorator
+
+def _get_permissible_items(request, applabel_modelname):
+    applabel, modelname = applabel_modelname.rsplit('_', 1)
+
+    typematched = []
+    for item in request.permissible_items or []:
+        if item['contenttype'] == applabel_modelname:
+            typematched.append(item)
+
+    # @todo -- Need to work `or_404` in
+    if not typematched:
+        return None
+    if len(typematched) > 1:
+        raise ValueError('Multiple obj or qs on the same type is not yet supported.')
+
+    type = typematched[0]
+    ctype = ContentType.objects.get_by_natural_key(applabel, modelname)
+    if type['kind'] == 'obj':
+        return ctype.get_object_for_this_type(**type['params'])
+    else:
+        return ctype.model_class().objects.filter(**type['params'])
+
+def _collect_args(fieldlookups, args):
+    results = {}
+    for lookup, arg_name in fieldlookups.iteritems():
+        if arg_name in args:
+            results[lookup] = args[arg_name]
+        else:
+            results[lookup] = arg_name
+
+    return results
+
+def permissible_object(func=None, contenttype=None, or_404=True, kind='obj',
+            fieldlookups_kwargs={}, fieldlookups_getparams={}, fieldlookups_postparams={}, **kwargs):
+    '''
+    Decorator for `permission` decorated (such as @permission_required) views for
+    object-level permission check. This decorators add `permissible_items` to the
+    request object.
+
+    Adapted from `django-trusts/decorators.py`
+    '''
+
+    def decorator(view_func):
+        @wraps(view_func, assigned=available_attrs(view_func))
+        def _wrapped_view(request, *args, **kwargs):
+            if not hasattr(request, 'permissible_items'):
+                request.permissible_items = []
+
+            params = {}
+            if fieldlookups_kwargs:
+                params.update(_collect_args(fieldlookups_kwargs, kwargs))
+
+            if fieldlookups_getparams:
+                params.update(_collect_args(fieldlookups_getparams, request.GET))
+
+            if fieldlookups_postparams:
+                params.update(_collect_args(fieldlookups_postparams, request.POST))
+
+            request.permissible_items.append({
+                'kind': kind,
+                'or_404': or_404,
+                'contenttype': contenttype,
+                'params': params,
+                'extra': kwargs
+            })
+
+            return view_func(request, *args, **kwargs)
+        return _wrapped_view
+    if func:
+        return decorator(func)
+    return decorator
+
+def permissible_queryset(func=None, contenttype=None, or_404=True,
+            fieldlookups_kwargs={}, fieldlookups_getparams={}, fieldlookups_postparams={}, **kwargs):
```

Comment:
None of the function you pass those dicts modify them? Even then I'd avoid using mutables as default args.